_posts/grant_guo/2017-07-07-spark.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title:  "A performance-tuning practice: from 20 hours to 8 minutes"
+breadcrumb: true
+author: grant_guo
+date: 2017-07-06
+categories: grant_guo
+tags:
+    - Spark SQL
+teaser:
+    info: An introduction of how to tune DCC-Release ID job and reduce the execution time greatly
+    image: spark-logo-trademark.png
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+## Background
+
+In our product environment, the DCC-Release ID job takes 20+ hours to finish. The bottleneck is the fact that for every data point ID job needs to talk to DCC-ID service through a HTTP request to either obtain or create an ID, which means every time the ID job runs there are millions of ID lookup requests, even if ID service runs multiple instances, and the vanish cache runs on a dedicated server with 128G RAM.
```

Comment:
You should break down this paragraph in shorter sentence, suggested version:

The bottleneck is the fact that, for every data point, the ID job needs to talk to DCC-ID service through a HTTP request to either obtain or create an ID. It means every time the ID job runs there are millions of ID lookup requests. This proves to be challenging, even if ID service runs multiple instances and the vanish cache runs on a dedicated server with 128G RAM.